crypto/asn1/a_gentm.c

```diff
@@ -88,6 +104,9 @@ int asn1_generalizedtime_to_tm(struct tm *tm, const ASN1_GENERALIZEDTIME *d)
      * digits.
      */
     if (a[o] == '.') {
+        if (strict)
+            /* RFC 5280 forbids fractional seconds */
+            goto err;
         if (++o > l)
             goto err;
```

Comment:
This check doesn't seem to make sense to me. After doing the `++o`, o is equal to the number of characters processed. if `o == l` it means we can't read any more characters. If `o > l` it means we already read 1 character too much. As far as I can see we always NUL terminate a string, so this is no problem, but then the above check `a[o] == '.'` would have failed. Am I missing something?